spring-cloud-dataflow-core/src/test/java/org/springframework/cloud/dataflow/core/dsl/ComposedTaskParserTests.java

```diff
@@ -0,0 +1,1172 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core.dsl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.ANDAND;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.ARROW;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.CLOSE_PAREN;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.COLON;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.GT;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.IDENTIFIER;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.LITERAL_STRING;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.LT;
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.OPEN_PAREN;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.springframework.cloud.dataflow.core.dsl.graph.Graph;
+
+/**
+ * Test the parser and visitor infrastructure. Check it accepts expected data and 
+ * correctly handles bad data. Check that the parsed out
+ * converts to/from the graph format that the UI can use.
+ *
+ * @author Andy Clement
+ */
+public class ComposedTaskParserTests {
+
+	private ComposedTaskNode ctn;
+
+	@Test
+	public void basics() {
+		Tokens tokens = new ComposedTaskTokenizer().getTokens("App1");
+		assertToken(IDENTIFIER, "App1", 0, 4, tokens.next());
+		tokens = new ComposedTaskTokenizer().getTokens("App1 && App2");
+		assertToken(IDENTIFIER, "App1", 0, 4, tokens.next());
+		assertToken(ANDAND, "&&", 5, 7, tokens.next());
+		assertToken(IDENTIFIER, "App2", 8, 12, tokens.next());
+		tokens = new ComposedTaskTokenizer().getTokens("< > -> ( )");
+		assertToken(LT,"<",0,1,tokens.next());
+		assertToken(GT,">",2,3,tokens.next());
+		assertToken(ARROW,"->",4,6,tokens.next());
+		assertToken(OPEN_PAREN,"(",7,8,tokens.next());
+		assertToken(CLOSE_PAREN,")",9,10,tokens.next());
+	}
+	
+	@Test
+	public void tokenStreams() {
```

Comment:
It is testing that the stream of tokens coming out of the tokenizer are in the right order. I think it is ok.